infra/aws/terraform/management-account/s3.tf

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Cost and Usage
+
+resource "aws_kms_key" "cur_bucket_key" {
+  enable_key_rotation = true
+  multi_region = true
+}
+
+resource "aws_kms_key" "cur_integrations_bucket_key" {
+  enable_key_rotation = true
+  multi_region = true
+}
+
+module "cur_reports_s3_bucket" {
+  source = "terraform-aws-modules/s3-bucket/aws"
+
+  bucket = "k8s-infra-cur-reports-bucket"
+  acl    = "private"
+
+  # S3 bucket-level Public Access Block configuration
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_policy = true
+  policy = data.aws_iam_policy_document.cur_reports_s3_bucket.json
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.cur_bucket_key.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+
+  versioning = {
+    enabled    = true
+    mfa_delete = false
+  }
+}
```

Comment:
This bucket should live in an AWS account in the infrastructure OU.
KMS encryption is nice to have but IMO not essential.

The only exception would be if CUR requires that the bucket is in the management account.

This is because we should minimize the amount of infrastructure that we define in the management account.

Reply:
Also, if we can make the bucket be `BucketOwnerEnforced`, let's do that (it turns off ACLs).

AWS' documentation isn't clear, but I have a feeling that CUR doesn't yet support `BucketOwnerEnforced` buckets. If that support is already in place, we should take advantage.

Reply:
> To ensure the bucket owner can access these reports, AWS further applies a BucketOwnerFullControl ACL to the reports when delivering them. By default, Amazon S3 objects such as these reports can only be read by the user or service principal who wrote them. To provide you or the bucket owner with permission to read the reports AWS must apply the BucketOwnerFullControl ACL. The ACL grants the bucket owner Permission.FullControl for these reports.

From: https://docs.aws.amazon.com/cur/latest/userguide/cur-s3.html

we should set the ACL to `log-delivery-write` so we don't apply ACLs at the object level.